Add documentation for restoring to Stateful Set

TehTotalPwnage opened the pull request:
This PR adds documentation for restoring volumes from the `Longhorn UI` in order to use them in Kubernetes `Stateful Sets`. The instructions have been split into two sections for `csi` and `flexvolume` drivers, and both at the end lead back to the same common instructions used to deploy the `Persistent Volume Claims` and the actual `Stateful Set`.

This PR resolves #166.

yasker commented on docs/restore_statefulset.md:
```diff
@@ -0,0 +1,208 @@
+2. In Kubernetes, create a `Persistent Volume` for each Longhorn volume that was
+created. Name the volumes something that can easily be referenced later for the
+`Persistent Volume Claims`:
+
```
`User must replace <size> , xxxx, and xxx in the yaml file below`

yasker commented on docs/restore_statefulset.md:
```diff
@@ -0,0 +1,208 @@
+  name: statefulset-vol-0
+spec:
+  capacity:
+    storage: 2Gi # must match size of Longhorn volume
```
Use e.g. `<>` in the yaml file so it won't work without filling in.

I'd love to let the user have the ability to restore the backup just by copy/paste/run the yaml file, but we don't know the backup URL.

yasker commented on docs/restore_statefulset.md:
```diff
@@ -0,0 +1,208 @@
+instructions:
+1. Connect to the `Longhorn UI` page in your web browser. Under the `Backup` tab,
+select the name of the `Stateful Set` volume. Click the dropdown menu of the
+volume entry and select `Get URL`.
```
`The URL would looks like xxxxxx`

yasker commented on docs/restore_statefulset.md:
```diff
@@ -0,0 +1,208 @@
+  - The name of the `Persistent Volume Claim` must follow this naming scheme:
+  `<name of Volume Claim Template>-<name of Stateful Set>-<index>`. Stateful
+  Set Pods are zero-indexed.
+
```
`In the following yaml, we assume the volume in the statefulset is "data" and the statefulset name is "webapp"`

TehTotalPwnage commented:
Changes:
- Added `<>` substitutions and documented replaceable data for `PersistentVolume` fields.
- Added backup URL example.
- Documented example `PersistentVolumeClaim` naming scheme.

yasker commented on docs/restore_statefulset.md:
```diff
@@ -0,0 +1,221 @@
+  name: statefulset-vol-0
+spec:
+  capacity:
+    storage: <> # must match size of Longhorn volume
```
I meant, `<size>`

TehTotalPwnage replied:
Whoops, fixing.
